### DISCUSSION TITLE
Best practice for asking a large number of questions.

### DISCUSSION THREAD

**Discussion by kingnobro** [acheong08/ChatGPT#571] [Q&A] [answered]
I have about 0.5 million questions to ask..

What is the best way to do this? Is there someone who has already implemented it? Thanks.

My concern are high latency and credit limitaion.

**Comment by acheong08** [comment, marked as answer]
> High latency

You can run multiple operations simultaneously via multi-threading. Python is not the best option for this due to high memory usage. You can port this library to Go since the Official API is available.

> Credit limitation

The model `text-chat-davinci-002-20221122` is completely free. Rate limit is 50 requests per minute. You can cycle multiple API keys to get around that.

**Reply by kingnobro** [reply]
Thanks for your fast reply!

**Comment by jigipatel7048** [comment]
@acheong08 , Where is official api document for converstion _id and parent_id?

**Reply by acheong08** [reply]
There is none. We have the base model but none of the fancy back end handling. Conversations are handled ourselves. You can check the code in `Official.py`
